packages/sdk-js/src/Identity.ts

```diff
@@ -0,0 +1,456 @@
+    criteria: {
+      verificationMethod?: DidUrl | UriFragment
+      verificationRelationship?: string
+      algorithm?: string
+    } = {}
+  ): SignerInterface {
+    const [signer] = this.getSigners(criteria)
+    if (typeof signer === 'undefined') {
+      throw new SDKErrors.NoSuitableSignerError(undefined, {
+        signerRequirements: criteria,
+        availableSigners: this.didSigners,
+      })
+    }
+    return signer
+  }
+}
+
+export type TransactionStrategy<T extends IdentityClass> = (
+  identity: IdentityClass & Identity
+) => Promise<T>
+
+export async function makeIdentity(args: {
+  did?: Did
+  didDocument?: DidDocument
+  signers?: SignerInterface[]
+  keypairs?: Array<Keypair | KeyringPair>
+  resolver?: typeof resolve
+}): Promise<IdentityClass>
+export async function makeIdentity<T extends IdentityClass>(args: {
+  did?: Did
+  didDocument?: DidDocument
+  signers?: SignerInterface[]
+  keypairs?: Array<Keypair | KeyringPair>
+  resolver?: typeof resolve
+  transactionStrategy: TransactionStrategy<T>
+}): Promise<T>
+export async function makeIdentity<T extends IdentityClass>({
+  did,
+  didDocument,
+  keypairs,
+  signers,
+  resolver = resolve,
+  transactionStrategy,
+}: {
+  did?: Did
+  didDocument?: DidDocument
+  signers?: SignerInterface[]
+  keypairs?: Array<Keypair | KeyringPair>
+  resolver?: typeof resolve
+  transactionStrategy?: TransactionStrategy<T>
+}): Promise<IdentityClass | T> {
+  let didDoc = didDocument
+  if (!didDoc) {
+    if (!did) {
+      throw new Error('either `did` or `didDocument` is required')
+    }
+    didDoc = await loadDidDocument(did, resolver)
+  }
+  const identity = new IdentityClass({
+    didDocument: didDoc,
+    signers,
+    resolver,
+  })
+  await identity.update({ skipResolution: true })
+  if (keypairs && keypairs.length !== 0) {
+    await identity.addKeypair(...keypairs)
+  }
+  if (!transactionStrategy) {
+    return identity
+  }
+  return transactionStrategy(identity)
+}
+
+type TypedKeyPair =
+  | (Keypair & { type: KeyringPair['type'] | 'x25519' })
+  | KeyringPair
+// | SignerInterface<Signers.DidPalletSupportedAlgorithms, Address>
+
+function isTypedKeyPair(
+  input: unknown
+): input is TypedKeyPair & NewDidVerificationKey {
+  return (
+    typeof input === 'object' &&
+    input !== null &&
+    'type' in input &&
+    'publicKey' in input &&
+    ['ed25519', 'sr25519', 'ecdsa'].includes((input as any).type) &&
+    ('secretKey' in input || 'sign' in input)
+  )
+}
+
+type TypedKeyPairs = {
+  authentication: [TypedKeyPair]
+  assertionMethod?: [TypedKeyPair]
+  delegationMethod?: [TypedKeyPair]
+  keyAgreement?: [TypedKeyPair]
+}
+
+export async function newIdentity(args: {
+  keys: TypedKeyPair | TypedKeyPairs
+  resolver?: typeof resolve
+}): Promise<IdentityClass>
+export async function newIdentity<
+  T extends IdentityClass &
+    Required<Pick<Identity, 'submitTx' | 'submitterAddress'>>
+>(args: {
+  keys: TypedKeyPair | TypedKeyPairs
```

Comment:
this parameter could accept signers too; but the name would have to change. I don't like it anyway though - it's too similar but different from the `keypairs` parameter of `makeIdentity`